File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    bool checkDivisibility(int n) {
        int sum = 0;
        int mul = 1;
        for (int i = n; i > 0; i /= 10) {
            int rem = i % 10;
            sum += rem;
            mul *= rem;
        }
        return n % (sum + mul) == 0;
    }
};

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.java
class Solution {
    public boolean checkDivisibility(int n) {
        int temp= n;
        int sum=0;
        int prod=1;
        while(temp>0){
            int x=temp%10;
            sum=sum+x;
            prod=prod*x;
            temp=temp/10;

        }
        int endsum=sum+prod;
        if(n%endsum==0){
            return true;
        }
        else{
            return false;
        }
    }
}